podaac/podaac.py

```diff
@@ -501,7 +507,32 @@ def granule_subset(self, input_file_path):
         token = result['token']
         conn.close()
 
-        return token
+        flag = 0
+        while(flag == 0):
+            url = url = self.URL + "subset/status?token=" + token
+            subset_response = requests.get(url).text
+            subset_response_json = json.loads(subset_response)
+            status = subset_response_json['status']
+            if (status == "done"):
+                flag = 1
+            if (status == "error"):
+                raise Exception(
+                    "Unexpected error occured for the subset job you have requested")
+            time.sleep(1)
+
+        print("Done! downloading the dataset zip .....")
+        download_url = subset_response_json['resultURLs'][0]
+        split = download_url.split('/')
+        length = len(split)
+        zip_file_name = split[length-1]
+        if path == '':
+            path = os.path.join(os.path.dirname(__file__), zip_file_name)
+        else:
+            path = path + zip_file_name
+        response = urlretrieve(download_url, path)
+        zip_content = zipfile.ZipFile(path)
+        z.extractall()
```

Comment:
Veriable Z does not exist yet. Also I wonder if we can create a small unit test? This is looking great BTW @Omkar20895 .